test/e2e/gitopsaddon/scripts/setup_env.sh

```
@@ -220,28 +220,47 @@ spec:
-    CTRL_POD=$(${KUBECTL} --context ${HUB_CTX} get pods -n open-cluster-management \
-      --no-headers -o name 2>/dev/null | grep multicloud-integrations-gitops | head -1)
-    if [ -n "${CTRL_POD}" ]; then
-      ${KUBECTL} --context ${HUB_CTX} logs -n open-cluster-management "${CTRL_POD}" --tail=50 || true
-    else
-      echo "  (controller pod not found — listing pods in open-cluster-management:)"
-      ${KUBECTL} --context ${HUB_CTX} get pods -n open-cluster-management || true
+echo ">> Waiting for argocd-agent secrets to be created by the controller"
+for secret_name in argocd-agent-resource-proxy-tls argocd-agent-jwt; do
+  echo ">> Waiting for ${secret_name}..."
+  for i in $(seq 1 60); do
+    if ${KUBECTL} --context ${HUB_CTX} -n openshift-gitops get secret "${secret_name}" &>/dev/null; then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)